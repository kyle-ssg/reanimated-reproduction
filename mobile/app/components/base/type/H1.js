import React, { Component } from 'react';
import propTypes from 'prop-types';

const h1 = class extends Component {
    static displayName = 'H1';

    render() {
      return (
          <Text
            accessible={this.props.accessible}
            accessibilityLabel={this.props.accessibilityLabel}
            style={[Styles.h1, this.props.style]}
            >
              {this.props.children}
          </Text>
      );
    }
};

h1.propTypes = {
  accessible: propTypes.bool,
  accessibilityLabel: propTypes.string,
  style: propTypes.any,
  children: propTypes.node,
};

module.exports = h1;
