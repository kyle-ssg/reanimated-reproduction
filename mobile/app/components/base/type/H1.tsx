import React from "react";
interface Props {
  accessible?: boolean;
  accessibilityLabel?: string;
  style?: ReactNative.TextStyle;
  children?: React.ReactNode;
}

const h1: React.FC<Props> = ({
  accessible,
  accessibilityLabel,
  style,
  children,
}) => (
  <Text
    accessible={accessible}
    accessibilityLabel={accessibilityLabel}
    style={[Styles.h1, style]}
  >
    {children}
  </Text>
);

export default h1;
