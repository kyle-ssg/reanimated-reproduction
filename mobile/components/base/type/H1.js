/**
 * Created by kylejohnson on 11/05/2016.
 */
/**
 * Created by kylejohnson on 04/05/2016.
 */
module.exports = class extends React.Component {
  render () {
    return (
      <Text style={[Styles.h1, this.props.style]}>{this.props.children}</Text>
    )
  }
};
