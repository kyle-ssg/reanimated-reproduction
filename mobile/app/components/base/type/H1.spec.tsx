import * as React from "react";
import H1 from "./H1";
import renderer from "react-test-renderer";


describe("<H1 />", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<H1>Example Title</H1>);
    expect(tree).toMatchSnapshot();
  });
});
