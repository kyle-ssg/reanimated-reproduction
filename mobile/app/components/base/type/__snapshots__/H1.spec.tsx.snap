// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "System",
        "fontSize": 32,
      },
      undefined,
    ]
  }
>
  Example Title
</Text>
`;
